src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -174,7 +174,116 @@ static status_t del_routes_from_routing_table(private_kernel_super_ipsec_t* this
 
 static void remove_ike_sa_matching(struct pm_terminator_conn* const terminator_conn);
 static void write_terminator_tunnels_oper(struct pm_terminator_conn* const terminator_conn);
+static struct pm_terminator_conn* lookup_terminator_conn(private_kernel_super_ipsec_t* this,
+														 host_t* const host_addr);
+static host_t* _get_sp_lookup_addr(const kernel_ipsec_manage_policy_t* const data,
+								   const kernel_ipsec_policy_id_t* const id);
+
+static void write_tunnel_oper_items(private_kernel_super_ipsec_t* this,
+									kernel_ipsec_manage_policy_t* data,
+									kernel_ipsec_policy_id_t* id,
+									ike_super_peer_info_t* super_peer_info,
+									child_sa_t * child_sa,
+									bool write_uptime) {
+	if(IKE_SUPER_PEER_INFO_TYPE_TUNNEL == super_peer_info->get_type(super_peer_info)) {
+		return;
+	}
+	ike_sa_t* const ike_sa = child_sa->get_ike_sa(child_sa);
+	ike_sa_id_t * const ike_sa_id = ike_sa->get_id(ike_sa);
+	struct oper_data oper_data;
+
+	uint16_t enc_alg = ENCR_UNDEFINED, int_alg = AUTH_UNDEFINED, size;
+	uint16_t esn = NO_EXT_SEQ_NUMBERS;
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), ENCRYPTION_ALGORITHM,
```

Comment:
Also, need to check the return value of this function. Nothing in the documentation it states what is the value of the out params in case no match is found, so you shouldn't count on the function not to touch this variable in any way. Instead explicitly set the value in case of an error